gbpservice/nfp/lifecycle_manager/drivers/lifecycle_driver_base.py

```diff
@@ -0,0 +1,440 @@
+                            % (device_data['compute_policy']))
+
+        try:
+            token = (device_data['token']
+                     if device_data.get('token')
+                     else self.identity_handler.get_admin_token())
+        except Exception:
+            LOG.error(_('Failed to get token for plug interface to device'
+                        ' operation'))
+            return False  # TODO[RPM]: should we raise an Exception here?
+
+        try:
+            for port in device_data['ports']:
+                port_id = self._get_port_id(port, token)
+                self.compute_handler_nova.attach_interface(
+                            token,
+                            device_data['tenant_id'],
+                            device_data['id'],
+                            port_id)
+        except Exception:
+            LOG.error(_('Failed to plug interface(s) to the device'))
+            return False  # TODO[RPM]: should we raise an Exception here?
+        else:
+            return True
+
+    def unplug_network_function_device_interfaces(self, device_data):
+        if not self.supports_hotplug:
+            raise Exception("Driver doesn't support interface hotplug")
+
+        if any(key not in device_data
+               for key in ['id',
+                           'tenant_id',
+                           'compute_policy',
+                           'ports']):
+            # TODO[RPM]: raise proper exception
+            raise Exception('Not enough required data is received')
+
+        # Not handling operations on compute policies other than 'nova' for now
+        if device_data['compute_policy'] != 'nova':
+            # TODO[RPM]: raise proper exception
+            raise Exception("Driver doesn't support operation"
+                            " with compute policy %s"
+                            % (device_data['compute_policy']))
+
+        try:
+            token = (device_data['token']
+                     if device_data.get('token')
+                     else self.identity_handler.get_admin_token())
+        except Exception:
+            LOG.error(_('Failed to get token for unplug interface from device'
+                        ' operation'))
+            return False  # TODO[RPM]: should we raise an Exception here?
+
+        try:
+            for port in device_data['ports']:
+                port_id = self._get_port_id(port, token)
+                self.compute_handler_nova.detach_interface(
+                            token,
+                            device_data['tenant_id'],
+                            device_data['id'],
+                            port_id)
+        except Exception:
+            LOG.error(_('Failed to unplug interface(s) from the device'))
+            return False  # TODO[RPM]: should we raise an Exception here?
+        else:
+            return True
+
+    def get_network_function_device_healthcheck_info(self, device_data):
```

Comment:
Can this be be merged with get_n._f.._d.._sharing_info()? What are the cons of merging this?